src/ProjectNeon/Assets/Features/Effects/Attack.cs

```diff
@@ -1,18 +1,42 @@
-﻿using System.Collections;
-using System.Collections.Generic;
+﻿
+
+
 using UnityEngine;
-using UnityEngine.Events;
 
-public sealed class Attack 
+public sealed class Attack  : Effect
 {
-    public Member Attacker { get; }
-    public Member Target { get; }
+    private Member _attacker;
+    public Member Attacker() { return this._attacker; }
+    private Target _target;
+    public Target Target() { return this._target; }
     public int Damage { get; }
 
-    public Attack(Member attacker, Member target, int damage)
+    public Attack(int damage)
     {
-        Attacker = attacker;
-        Target = target;
         Damage = damage;
     }
+
+    public void Apply(Member source, Target target)
+    {
+        _attacker = source;
+        _target = target;
+        if (target.Members.Length > 1)
+        {
+            target.Members.ForEach(
+                member => {
+                    new Attack(Damage).Apply(source, target);
+                }
+            );
+        } else
+        {
+            AllEffects.Create(
```

Comment:
Avoid circular usage references, where `AllEffects` depends on `Attack` which depends on `AllEffects`.

Reply:
@SilasReinagel please see above

Reply:
This is a critical issue. We can't merge unless this is fixed.